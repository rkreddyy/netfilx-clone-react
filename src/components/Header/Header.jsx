import React from "react";
import Logo from "../common/Logo";
import "./Header.scss";

export default ({ showAddModal }) => (
  <div className="Header">
    <div className="Header-content">
      <div className="top-panel">
        <div className="top-panel-label">
          <Logo />
        </div>
        <button
          className="btn-add-movie"
          onClick={showAddModal}
        >
          +ADD MOVIE
          </button>
      </div>
      <div className="search-panel">
        <span className="search-panel-label">FIND YOUR MOVIE</span>
        <input
          className="search-panel-input"
          type="text"
          placeholder="What do you want to watch?"
        />
        <button className="search-panel-button">SEARCH</button>
      </div>
    </div>
  </div>
);
