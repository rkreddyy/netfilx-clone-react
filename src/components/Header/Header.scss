@import "../../assets/scss/colors";
@import "../../assets/scss/mixins";

.Header {
  padding: 1em 3em;
  background-image: linear-gradient(rgba(0, 0, 0, 0.6), rgba(0, 0, 0, 0.6)), url("../../assets/img/netflixlist.jpeg");
  background-position: center;
  background-size: 50% 100%;

  &-content {
    .top-panel {
      display: flex;
      justify-content: space-between;

      .btn-add-movie {
        cursor: pointer;
        font-size: 1.3em;
        border: none;
        border-radius: 5px;
        font-weight: 500;
        background-color: rgba(181, 181, 181, .4);
        padding: 8px 20px;
        letter-spacing: 1px;
        color: $pink;

        &:hover {
          background-color: rgb(89, 89, 89, .4);
        }

        &:active {
          background-color: rgb(89, 89, 89, .1);
        }

        &:focus {
          outline: none;
        }
      }
    }

    .search-panel {
      display: flex;
      flex-wrap: wrap;
      margin: 20px 60px 120px;

      &-label {
        color: $white;
        width: 100%;
        font-size: 3em;
        font-weight: 100;
        margin-bottom: 30px;
        letter-spacing: 2px;
      }

      &-input {
        color: $white;
        background-color: rgba(31, 31, 31, 0.6);
        flex: 2;
        font-size: 20px;
        padding: 10px;
        width: 50%;
        border: none;
        border-radius: 5px;
      }

      &-button {
        @include btn(
          250px,
          $white,
          none,
          $pink,
          $pink-light,
          $pink-dark
        );
      }
    }
  }
}