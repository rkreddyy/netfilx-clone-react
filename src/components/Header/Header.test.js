import Header from './Header';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

describe('Header', () => {
    test('Should render Header', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Header />, div);
        expect(div.textContent).toBe('React JS Global Mentoring Program');
    });

    test('Should match the snapshot', () => {
        const component = renderer.create(<Header />).toJSON();
        expect(component).toMatchSnapshot();
    });
});
