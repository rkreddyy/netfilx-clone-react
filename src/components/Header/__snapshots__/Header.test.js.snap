// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should match the snapshot 1`] = `
<h1>
  React JS Global Mentoring Program
</h1>
`;
